python/hyperspectral/spectra/spectra.py

```diff
@@ -0,0 +1,203 @@
+        return Spectrum(self.name, self.type_, self.class_, min(x), max(x), len(x), deepcopy(self.info), x ,y)
+
+    def __getitem__(self, ix):
+        return self.info[ix]
+
+    @staticmethod
+    def parse_ECOSTRESS(filename):
+        spaces = parsec.regex(r'\s*', re.MULTILINE)
+
+        @parsec.generate
+        def header_field():
+            k = yield spaces >> parsec.regex('[^:]*') << spaces
+            yield parsec.string(':')
+            v = yield parsec.ends_with(parsec.regex('[^\n]*'), parsec.string('\n'))
+            return { k: v.strip() }
+
+        @parsec.generate
+        def header():
+            items = yield parsec.many(header_field)
+            d = {}
+            for item in items:
+                d.update(item)
+            return d
+
+        floating = parsec.regex('[-+]?([0-9]+(\.[0-9]+)?|\.[0-9]+)')
+
+        @parsec.generate
+        def sample():
+            fwhm = yield spaces >> floating << spaces
+            level = yield floating << spaces
+            yield parsec.optional(parsec.string('\n'))
+            return (float(fwhm), float(level))
+
+        @parsec.generate
+        def parser():
+            head = yield header
+            yield parsec.many(parsec.string('\n'))
+            samps = yield parsec.many(sample)
+            return head, samps
+
+        def parse_spectrum(filename):
+            try:
+                with open(filename, 'r', encoding='iso-8859-1') as f:
+                    parsed = parser.parse(f.read())
+                return parsed
+            except:
+                print('Error parsing '+filename)
+                raise
+
+        header, data = parse_spectrum(filename)
+
+        name = header.pop('Name')
+        type_ = header.pop('Type')
+        class_ = header.pop('Class')
+        x0 = float(header.pop('First X Value'))
+        x1 = float(header.pop('Last X Value'))
+        x_min, x_max = min(x0, x1), max(x0, x1)
+        n_samples = header.pop('Number of X Values')
+
+        x = list(map(lambda v: v[0], data))
+        y = list(map(lambda v: v[1], data))
+
+        return Spectrum(name, type_, class_, x_min, x_max, n_samples, header, x, y)
+
+
+class SpectralLibrary:
+    def __init__(self, spectra):
+        self.spectra = spectra
+        self.grp_fn = None
+        self.groups = None
+        self.source_bands = list(range(0, len(spectra[0].x)))
+
+    def is_regular(self, tol=1e-8):
+        m = np.array([s.x for s in self.spectra])
+        return len(m.shape) == 2 and all(np.abs(np.min(m, axis=0) - np.max(m, axis=0)) < tol)
+
+    def regularize(self, band_frequencies, source_bands=None):
+        self.spectra = [s.resample(band_frequencies) for s in self.spectra]
+        if source_bands:
+            assert len(source_bands) == len(band_frequencies)
+            self.source_bands = source_bands
+        else:
+            self.source_bands = list(range(0, len(band_frequencies)))
+
+    @property
+    def bands(self):
```

Comment:
Looks like this is returning a count of bands? If so might want to rename to `band_count` for consistency with `def invalid_band_count`